### ISSUE TITLE
[Search field] "Clear text" button is not keyboard accessible

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3132]
## Current behavior 

Keyboard user can place their focus only inside search `input` but cannot place it on "Clear text" `button` because it has `tabindex=-1`. Screen reader user can place screen reader's focus on this button using arrows but not using `Tab`. 
This behavior is mimicking `input type='search'` in which "Clear" `button` cannot be reached via keyboard nor screen reader. But this button is not presented in some browsers like Firefox and its behavior is not considered accessible: https://design-patterns.service.justice.gov.uk/components/search/#accessibility-issues  

<img width="393" height="67" alt="Image" src="https://github.com/user-attachments/assets/b41c546d-a716-456e-966a-f7d48d4609d5" />

## Suggested solution

It's recommended to change `tabindex` value from `-1` to `0` and make sure focus indicator is visible and it has sufficient color ratio of at least 3:1 between background.  That should make button fully accessible for keyboard users.

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
